### ISSUE TITLE
Start balancing spaces only when non space characters are present

### ISSUE THREAD

**Issue by feekApp** [altermo/ultimate-autopair.nvim#20]
Hi,

I have a suggestion for starting the balancing spaces. Reason for the request are the Markdown ToDo  items:

``` md

- [ ] ToDo item
- [ ] 

```

When creating a ToDo item Ultimate Autopairs does the following (`_` is a space and `|` is the cursor):

```
- [|]
pressing space in order to generate empty task:
-[_|_]

wanted behaviour:
- [_|]

```

It would be nice you  start the balancing spaces only when 'non space' characters are present inside the autopair:

```
(|)
pressing space:
(_|)
pressing an other space:
(__|)
typing non space character (means fire up the auto balance module) :
(__A|__)
```

== Feek

**Comment by altermo** [comment]
First, the wanted behavior is already implemented (only in markdown, not in other markup languages).
Second, the idea seems logical, but detecting ALL inserted characters is harder (using `vim.on_key`), as there are so many exceptions to remember like whether the previous keypress was `<C-v>` or `<C-k>` (and that it was in insert mode).
Still, as there are only a dozen exceptions to keep track of, it is possible to implement.
In conclusion: I may implement it in the future.

**Comment by feekApp** [comment]
Totally clear!! All those exceptions makes implementation hard (or even impossible). 
Now I notice that's it indeed already implemented in Markdown, but because I'm also using VimWiki, the action is was not recognised (I use `.md` files for wiki).

== Feek
